### PR TITLE
Jl/refactor deploy params

### PR DESCRIPTION
**Pull request by jordlay**
---
Refactored deploy params to make less strictly matching schema and values. We now search for the deployParams in the values.mappings file and then separately search for those in the schema to find their type.

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py [hunk 1]
```diff
@@ -476,94 +476,71 @@ def get_chart_mapping_schema(
 
         with open(values_schema, "r", encoding="utf-8") as f:
             data = json.load(f)
-            schema_data = data["properties"]
+            schema_data = data
```
Comment (jddarby) [inline src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py:479]: Can we just change `data` to `schema_data`? i.e. `schema_data = json.load(f)`

File: src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py [hunk 2]
```diff
@@ -476,94 +476,71 @@ def get_chart_mapping_schema(
...
 
         try:
-            final_schema = self.find_deploy_params(values_data, schema_data, {})
+            deploy_params_dict = self.traverse_dict(values_data, DEPLOYMENT_PARAMETER_MAPPING_REGEX)
+            new_schema = self.search_schema(deploy_params_dict, schema_data)
         except KeyError as e:
             raise InvalidTemplateError(
-                f"ERROR: Your schema and values for the helm package '{helm_package.name}' do not match. \
+                f"ERROR: There is a problem with your schema or values for the helm package '{helm_package.name}'. \
```
Comment (jddarby) [inline src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py:486]: Can we add something a bit more explicit here? Like what is the key we failed to find the schema for?

Comment (jordlay) [reply src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py:486]: we have relaxed the rules about matching schemas here so i think a more generic answer is ok for now, people are unlikely to hit this. I could just remove the try except completely tbh

File: src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py [hunk 3]
```diff
@@ -476,94 +476,71 @@ def get_chart_mapping_schema(
...
+        while stack:  # While there are still items in the stack
+            # Pop the last item from the stack and unpack it into node (the dictionary) and path
+            (node, path) = stack.pop()
+            for k, v in node.items():   # For each key-value pair in the popped item
+                if isinstance(v, dict):  # If the value is a dictionary
+                    stack.append((v, path + [k]))  # Add the dictionary to the stack with the path
+                elif isinstance(v, str) and re.search(target, v):  # If the value is a string + matches target regex
+                    match = re.search(target, v)  # Take the match i.e, foo from {deployParameter.foo}
+                    result[match.group(1)] = path + [k]  # Add it to the result dictionary with its path as the value
+            if isinstance(node, list):  # If the popped item is a list
```
[the remark this answers is not in the document]
Comment (jordlay) [reply src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py]: done

**Pull request merged**